File: Code/0088_Merge_Sorted_Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int _m = 0;
        int _n = 0;
        
        queue<int> q;
        
        while(_m < m || _n < n)
        {
            //cout << "_m:: " << _m << " _n:: " << _n << endl;
            if(_m == m)
            {
                q.push(nums2[_n]);
                _n++;
            }
            else if(_n == n)
            {
                q.push(nums1[_m]);
                _m++;
            }
            else if(nums1[_m] <= nums2[_n])
            {
                q.push(nums1[_m]);
                _m++;
            }
            else if(nums1[_m] > nums2[_n])
            {
                q.push(nums2[_n]);
                _n++;
            }
        }
        
        int i = 0;
        while(!q.empty())
        {
            nums1[i++] = q.front();
            q.pop();
        }
    }
};

File: Code/0088_Merge_Sorted_Array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        LinkedList<Integer> list = new LinkedList<Integer>();
        
        int im = 0;
        int in = 0;
        while(im < m || in < n){
            if(im == m)
                list.add(nums2[in++]);
            else if(in == n)
                list.add(nums1[im++]);
            else{
                if(nums1[im] < nums2[in])
                    list.add(nums1[im++]);
                else
                    list.add(nums2[in++]);
            }
        }

        im = 0;
        while(list.peek() != null){
            nums1[im++] = list.remove();
        }
    }
}